Rt685 flash remove code location

EmilioCBen opened the pull request:
Removed the CODE_LOCATION choice as part of the flexspi clock init setup and 
replaced the usage with the Kconfig FLASH_MCUX_FLEXSPI_XIP instead.

carlescufi commented:
@dleach02 @danieldegrasse @mmahadevan108 please review

danieldegrasse commented on soc/arm/nxp_imx/Kconfig:
```diff
@@ -23,4 +23,14 @@ config SOC_PART_NUMBER
+	depends on (CODE_FLEXSPI || CODE_FLEXSPI2 || SOC_SERIES_IMX_RT6XX)
+	select XIP
+	help
+	  Allows using the flash API while running in XIP.
```
We need to update the help string for this Kconfig, as it is no longer is specific to the flash driver

danieldegrasse commented on soc/arm/nxp_imx/Kconfig:
```diff
@@ -23,4 +23,15 @@ config SOC_PART_NUMBER
 
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	default $(dt_node_has_compat, $(DT_CHOSEN_FLASH_PARENT), $(DT_COMPAT_FLEXSPI))
```
We still need this KConfig to be enabled for the RT1xxx series. I have not verified that the `zephyr,flash` chosen node is the 
SPI flash chip for all RT boards. If this KConfig isn't set, flash samples will likely fail, as the flash driver will attempt to reconfigure the FlexSPI while running in XIP, which will likely crash the application. Have you tested this change with a flash sample on a R6xx, RT5xx, and RT1xxx board?

EmilioCBen replied:
I have not tested these SOC's yet, I can test real quick with the RT595

EmilioCBen committed:
boards: arm: rtxxx: moving the instances FLASH_MCUX_FLEXSPI_XIP

This Kconfig is moved to the soc level since it determines
the flexspi clock initialization for XIP.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
soc: arm: nxp_imx: replaced CODE_LOCATION and added FLEXSPI_XIP to soc

Cleaning up the instances of CODE_LOCATION used in the soc
clock_init and replaced them with the Kconfig
FLASH_MCUX_FLEXSPI_XIP due to the correlation with
the flexspi clocks and the XIP feature of Flexspi.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>
